### ISSUE TITLE
Error getting access token from refresh token

### ISSUE THREAD

**Issue by biswajit-paul** [thephpleague/oauth2-client#775]
When I am trying to get access token from saved refreshed token, I am getting this error:

```
<b>Fatal error</b>:  Uncaught TypeError: Argument 1 passed to League\OAuth2\Client\Provider\AbstractProvider::prepareAccessTokenResponse() must be of the type array, string given, called in /public_html/support/vendor/league/oauth2-client/src/Provider/AbstractProvider.php on line 529 and defined in /public_html/support/vendor/league/oauth2-client/src/Provider/AbstractProvider.php:708
Stack trace:
#0 /public_html/support/vendor/league/oauth2-client/src/Provider/AbstractProvider.php(529): League\OAuth2\Client\Provider\AbstractProvider-&gt;prepareAccessTokenResponse('This Basecamp I...')
#1 /public_html/support/upload.php(15): League\OAuth2\Client\Provider\AbstractProvider-&gt;getAccessToken(Object(League\OAuth2\Client\Grant\RefreshToken), Array)
#2 {main}
  thrown in <b>/public_html/support/vendor/league/oauth2-client/src/Provider/AbstractProvider.php</b> on line <b>708</b><br />
```

My code to get access token:

```
$grant = new \League\OAuth2\Client\Grant\RefreshToken(); 

$provider = new \League\OAuth2\Client\Provider\GenericProvider([
    'clientId'                => getenv('CLIENT_ID'),
    'clientSecret'            => getenv('CLIENT_SECRET'),
    'redirectUri'             => getenv('REDIRECT_URI'),
    'urlAuthorize'            => getenv('AUTH_URL'),
    'urlAccessToken'          => getenv('TOKEN_URL'),
    'urlResourceOwnerDetails' => ''
]);

$token = $provider->getAccessToken( $grant, [
    'refresh_token' => $refreshToken,
    'type'          => 'refresh'
] );
```

Thanks for help in advanced.

**Comment by ramsey** [comment]
Sorry for the late response.

This was due to a bug in league/oauth2-client version 2.3.0 that was fixed in version 2.3.1.

I recommend upgrading to the latest version, if you haven't already.

**Comment by ramsey** [comment]
For future reference (if anyone wants to understand my reply above), here is where this bug was fixed: https://github.com/thephpleague/oauth2-client/commit/9297f3bae37b8d7bdb876a0879cc3839ccc211b7#diff-e8490e4bb8acb102745699d2bd7aa0a298d836c92d00d2ed57dd4c7ad8b24282
